sticky-rows/StickyRows.js

```diff
@@ -0,0 +1,279 @@
+    }
+
+    if (config.error) {
+        throw new Error(config.error);
+    }
+}
+
+
+/** Uninstall triggers. */
+function uninstall() {
+    TriggerUtil.uninstall(TRIGGER_HANDLER_FUNCTION);
+}
+
+
+/** Install spreadsheet onChange trigger. */
+function install() {
+
+    TriggerUtil.uninstall(TRIGGER_HANDLER_FUNCTION);
+
+    Logger.log("Installing onChange trigger for spreadsheet %s", spreadsheet.getId());
+
+    const sheet = SpreadsheetApp.getActive();
+    ScriptApp.newTrigger(TRIGGER_HANDLER_FUNCTION)
+        .forSpreadsheet(SpreadsheetApp.getActiveSpreadsheet().getId())
+        .onChange()
+        .create();
+    Logger.log("Installed onChange trigger for spreadsheet %s", sheet.getSpreadsheet());
+}
+
+
+/** Perform the actual sticky rows logic.
+ *
+ * Supposed to be called in a SpreadSheet.onChange() trigger.
+ */
+function onChangeStickyRows(event) {
+
+    const spreadsheet = event.source || SpreadsheetApp.getActiveSpreadsheet();
+    if (!spreadsheet) {
+        throw new Error("Must run in the context of a Spreadsheet");
+    }
+
+    let sheet;
+    try {
+        sheet = spreadsheet.getActiveRange().getSheet();
+    } catch (e) {
+        sheet = spreadsheet.getActiveSheet();
+    }
+
+    const key = "stickyRows." + sheet.getSheetId();
+    const config = mergeConfig_(key, {});
+    if (!config.dynamic_range || !config.sticky_range || !config.controlCell) {
+        // STICKY_ROWS() also reports this error, we can't forward since we know no control cell here
+        throw new Error("Please call STICK_ROWS() with a valid dynamic_range and sticky_range in a formula on the sheet");
+    }
+
+    try {
+        realignStickyRows_(sheet, config);
+        if (config.error) {
+            forceRecalculation_(sheet, mergeConfig_(key, {error: ''}));
+        }
+    } catch (e) {
+        // store error message for displaying by STICKY_ROWS()
+        const nextError = '' + e;
+        if (nextError !== config.error) {
+            forceRecalculation_(sheet, mergeConfig_(key, {error: nextError}));
+        }
+        throw e;
+    }
+}
+
+
+/** Read, merge and update config properties.
+ *
+ * @return The updated config for the given key (sheet).
+ */
+function mergeConfig_(key, configProperties) {
+    const config = JSON.parse(PropertiesService.getDocumentProperties().getProperty(key) || '{}');
+    let changed = false;
+    for (const property in configProperties) {
+        if (config[property] !== configProperties[property]) {
+            config[property] = configProperties[property];
+        }
+        changed = true;
+    }
+
+    if (changed) {
+        PropertiesService.getDocumentProperties().setProperty(key, JSON.stringify(config));
+    }
+
+    return config;
+}
+
+
+/** Logic to re-align rows in the sticky area (defined in config) within sheet. */
+function realignStickyRows_(sheet, config) {
```

Comment:
Nitpick: Shouldn't the function name be `reAlignStickyRows_` following convention so far in this file.